stf.test/build.gradle

```diff
@@ -21,37 +25,58 @@ dependencies {
-}
 
-// TODO move into buildSrc
-task stfTest (type: Test, dependsOn: [':saros.eclipse:build']) {
-    group 'Verification'
-    description 'Runs the stf tests. Requires a corresponding test environment'
-
-    testClassesDirs = sourceSets.stfTest.output.classesDirs
+    stfFlakyAndFailingTestCompile project(':saros.stf')
+    stfFlakyAndFailingTestCompile project(path: ':saros.stf', configuration: 'testing')
+    stfFlakyAndFailingTestCompile configurations.testConfig
+    stfFlakyAndFailingTestCompile configurations.implementation
+    testCompile configurations.stfFlakyAndFailingTestCompile // for running the tests in eclipse
+}
 
+class StfTest extends Test {
+  StfTest() {
+    this.setGroup('Verification')
+    this.setDescription('Runs the stf tests. Requires a corresponding test environment')
     systemProperty 'saros.stf.client.configuration.files', System.getProperty('stf.client.configuration.files', '')
 
-    classpath = sourceSets.stfTest.runtimeClasspath
-
-    testLogging {
-        showStandardStreams = true
-        events "started", "passed", "skipped", "failed", "standardOut", "standardError"
+    this.testLogging{ loggingContainer ->
+        loggingContainer.setShowStandardStreams(true)
+        loggingContainer.setEvents(['started', 'passed', 'skipped', 'failed', 'standardOut', 'standardError'])
     }
-    // Print summary of failed tests
-    ext.failedStfTests = []
-    afterTest { descriptor, result ->
-        if(result.resultType == org.gradle.api.tasks.testing.TestResult.ResultType.FAILURE){
-            ext.failedStfTests << "${descriptor.className}.${descriptor.name}"
+
+    List<String> failedTests = new ArrayList<String>();
+    this.afterTest { descriptor, result ->
+        if( result.resultType == org.gradle.api.tasks.testing.TestResult.ResultType.FAILURE ){
+            failedTests << "${descriptor.className}.${descriptor.name}"
         }
     }
-    afterSuite { suite, result ->
-        if (!suite.parent && !ext.failedStfTests.isEmpty()) {
+
+    this.afterSuite { suite, result ->
+        if ( !suite.parent && !failedTests.isEmpty() ) {
             logger.lifecycle("Failed tests:")
-            ext.failedStfTests.each { logger.lifecycle(it) }
+            failedTests.each { logger.lifecycle(it) }
         }
     }
+  }
+}
 
+task stfTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
+    testClassesDirs = sourceSets.stfTest.output.classesDirs
+    classpath = sourceSets.stfTest.runtimeClasspath
     include '**/stf/test/**/*Test.*'
     exclude '**/stf/test/stf/**'
+
+    useJUnit {
+        excludeCategories 'saros.stf.test.categories.FlakyTests', 'saros.stf.test.categories.FailingTests'
+    }
 }
+
+task stfFlakyAndFailingTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
+    testClassesDirs = sourceSets.stfFlakyAndFailingTest.output.classesDirs
+    classpath = sourceSets.stfFlakyAndFailingTest.runtimeClasspath
+    ignoreFailures = true
+    include '**/stf/test/**/*Test.*'
```

Comment:
Is it intentional that the normal STF execution excludes the stf self-tests (see l.67) and the flaky run does not?

Reply:
It was not intended to have different exclude/include, but it is not necessary to have the exclude in this task, because there are no self-tests using categories. I can add the exclude (might be useful if we would use categories in the future in the self-tests).